pkg/checks/checks.go

```diff
@@ -54,68 +56,193 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 	}, ctx.Done())
 }
 
-func checkNamespace(namespace string) bool {
-	if namespace == "default" {
-		return true
-	}
+type Node struct {
+	Ready bool
+	Name  string
+}
 
-	whitelistedNamespaces := []string{
-		"openshift-",
-		"kube-",
-	}
-	for _, ns := range whitelistedNamespaces {
-		if strings.HasPrefix(namespace, ns) {
-			return true
+func GenerateNodeMap(cs *acsapi.OpenShiftManagedCluster) map[string][]Node {
+	nodes := make(map[string][]Node)
+	for _, app := range cs.Properties.AgentPoolProfiles {
+		nodes[string(app.Role)] = []Node{}
+		for i := 0; i < app.Count; i++ {
+			n := Node{Name: fmt.Sprintf("%s-%06d", app.Role, i)}
```

Comment:
I was probably wrong.  We can't do this.  There's no guarantee that node numberings are contiguously 0-N.